awswrangler/catalog/_create.py

```diff
@@ -501,16 +505,25 @@ def create_database(
     ...     name='awswrangler_test'
     ... )
     """
-    args: Dict[str, str] = {}
     client_glue: boto3.client = _utils.client(service_name="glue", session=boto3_session)
-    args["Name"] = name
+    args: Dict[str, str] = {"Name": name}
     if description is not None:
         args["Description"] = description
 
-    if catalog_id is not None:
-        client_glue.create_database(CatalogId=catalog_id, DatabaseInput=args)
-    else:
-        client_glue.create_database(DatabaseInput=args)
+    try:
+        r = client_glue.get_database(Name=name)
+        if not exist_ok:
+            raise exceptions.AlreadyExists(f"Database {name} already exists and <exist_ok> is set to False.")
+        if description and description != r["Database"].get("Description", ""):
+            if catalog_id is not None:
+                client_glue.update_database(CatalogId=catalog_id, Name=name, DatabaseInput=args)
+            else:
+                client_glue.update_database(Name=name, DatabaseInput=args)
+    except client_glue.exceptions.EntityNotFoundException:
+        if catalog_id is not None:
+            client_glue.create_database(CatalogId=catalog_id, DatabaseInput=args)
+        else:
+            client_glue.create_database(DatabaseInput=args)
```

Comment:
Should we simplify this to:
```
        if description and description != r["Database"].get("Description", ""):
            client_glue.update_database(
                **_catalog_id(catalog_id=catalog_id, Name=name, DatabaseInput=args)
            )
    except client_glue.exceptions.EntityNotFoundException:
        client_glue.create_database(
            **_catalog_id(catalog_id=catalog_id, DatabaseInput=args)
        )
```
`**_catalog_id` method will fetch the catalog_id if it exists and ignore otherwise

Reply:
Sounds good to me. I will change it